src/main/java/com/artipie/metrics/MetricsContext.java

```diff
@@ -0,0 +1,300 @@
+                                    ),
+                                    interval(entry.getValue())
+                                );
+                                break;
+                            default:
+                                throw new IllegalStateException(
+                                    String.format(
+                                        "%s is not interval type of metrics output",
+                                        entry.getValue()
+                                    )
+                                );
+                        }
+                        return res;
+                    }
+                ).toList();
+            this.publisher = new MetricsPublisher(this.metrics, list);
+        }
+    }
+
+    /**
+     * Metrics are enabled.
+     *
+     * @return True, if metrics are enabled.
+     */
+    public boolean enabled() {
+        return !this.outputs.isEmpty();
+    }
+
+    /**
+     * Log metrics output is enabled.
+     *
+     * @return True, if enabled.
+     */
+    public boolean logEnabled() {
+        return this.outputs.containsKey(MetricsOutputType.LOG);
+    }
+
+    /**
+     * Storage metrics output is enabled.
+     *
+     * @return True, if enabled.
+     */
+    public boolean storageEnabled() {
+        return this.outputs.containsKey(MetricsOutputType.ASTO);
+    }
+
+    /**
+     * Creates storage for metrics.
+     *
+     * @return Storage.
+     */
+    public Storage metricsStorage() {
+        if (!this.storageEnabled()) {
+            throw new IllegalStateException("Storage type metrics are not defined");
+        }
+        return metricsStorage(this.outputs.get(MetricsOutputType.ASTO)
+            .yamlMapping(MetricsContext.STORAGE)
+        );
+    }
+
+    /**
+     * Prometheus metrics are enabled.
+     *
+     * @return True, if Prometheus metrics are enabled.
+     */
+    public boolean prometheusEnabled() {
+        return this.outputs.containsKey(MetricsOutputType.PROMETHEUS);
+    }
+
+    /**
+     * Vertx metrics are enabled.
+     *
+     * @return True, if vertx metrics are enabled.
+     */
+    public boolean vertxEnabled() {
+        return this.outputs.containsKey(MetricsOutputType.VERTX);
+    }
+
+    /**
+     * Endpoint to expose vertx metrics.
+     *
+     * @return Endpoint, if {@code null} then vertx metrics are not enabled.
+     */
+    public String vertxEndpoint() {
+        String res = null;
+        if (this.vertxEnabled()) {
+            res = this.outputs.get(MetricsOutputType.VERTX).string("endpoint");
+        }
+        return res;
+    }
+
+    /**
+     * Port to expose vertx metrics.
+     *
+     * @return Port, if -1 then vertx metrics are not enabled.
+     */
+    public int vertxPort() {
+        int res = -1;
+        if (this.vertxEnabled()) {
+            res = Integer.parseInt(
+                this.outputs.get(MetricsOutputType.VERTX).string("port")
+            );
```

Comment:
Suggesting to replace three methods with one, which will return `Optional<Pair<String, Integer>>` as these three methods are not consistent: calling `vertxPort()` does not make sense if such metric is disabled, `vertxEndpoint()` returns `null` with is not good and etc.

Reply:
This is a temporary solution. 
In the result of #1175 we will not have separate parameters host/port for Vert.x metrics.